cmd/main.go

```diff
@@ -265,3 +260,58 @@ func getOutputFn(outputFlag *string, currentFn selector.InstanceTypesOutputFn) s
 	}
 	return outputFn
 }
+
+func getRegionAndProfileAWSSession(regionName *string, profileName *string) *session.Session {
+	sessOpts := session.Options{}
+	if regionName != nil {
+		sessOpts.Config.Region = regionName
+	}
+
+	if profileName != nil {
+		if sessOpts.Config.Region == nil {
+			if profileRegion, err := getProfileRegion(*profileName); err != nil {
+				log.Println(err)
+			} else {
+				sessOpts.Config.Region = &profileRegion
+			}
+		}
+		sessOpts.Profile = *profileName
+	}
+
+	sess := session.Must(session.NewSessionWithOptions(sessOpts))
+	if sess.Config.Region != nil && *sess.Config.Region != "" {
+		return sess
+	}
+	if defaultProfileRegion, err := getProfileRegion(defaultProfile); err == nil {
+		sess.Config.Region = &defaultProfileRegion
+	}
+	if sess.Config.Region == nil || *sess.Config.Region == "" {
+		if defaultRegion, ok := os.LookupEnv(defaultRegionEnvVar); ok && defaultRegion != "" {
+			sess.Config.Region = &defaultRegion
+		}
+	}
+	return sess
+}
+
+func getProfileRegion(profileName string) (string, error) {
+	if profileName != defaultProfile {
+		profileName = fmt.Sprintf("%s %s", "profile", profileName)
```

Comment:
is this better than writing `fmt.Sprintf("profile %s", profileName)` ? Also, why not print when it's the default profile?

Reply:
haha nope, `fmt.Sprintf("profile %s", profileName)` is way clearer :)

But it's not printing, it's creating a string with the static string "profile" prepended. The AWS config file prepends profile for custom profiles but does not for the "default" profile.  

Here's an example file:

```
cat ~/.aws/config
[default]
region = us-east-1
output = json
[profile jason]
region = us-east-2
output = json
[profile test]
output = json
```